rosbag2_storage_sqlite3/include/rosbag2_storage_sqlite3/sqlite_storage.hpp

```diff
@@ -126,7 +126,7 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
   int get_last_rowid();
   int read_db_schema_version();
   uint64_t get_page_size() const;
-  uint64_t read_total_page_count_locked() const RCPPUTILS_TSA_GUARDED_BY(db_read_write_mutex_);
+  uint64_t read_total_page_count_locked() const;
```

Comment:
I think `read_total_page_count_locked` actually requires `db_read_write_mutex_` lock since it queries SQL?

```suggestion
  uint64_t read_total_page_count_locked() const RCPPUTILS_TSA_REQUIRES(db_read_write_mutex_);
```

Reply:
But it never takes the lock: https://github.com/ros2/rosbag2/blob/931bf541a3547dd7d1bcec119d68cb95d7409a72/rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp#L881-L893

Oh, but you are saying maybe it *should* take the lock?  Yeah, that's possible.  @MichaelOrlov what do you think?

Reply:
@fujitatomoya The function itself shouldn't take a lock, however, it should rather be called only when the lock has already been acquired.
This is why I added `_locked` prefix to the name of the function and TSA annotation ` RCPPUTILS_TSA_REQUIRES(db_read_write_mutex_);` to make sure that this function will ever be called when the relevant mutex is already locked.

BTW. We have another function a few lines above in the header file with exactly the same TSA guard lock and behavior and no warnings.
```
void write_locked(std::shared_ptr<const rosbag2_storage::SerializedBagMessage> message)
  RCPPUTILS_TSA_REQUIRES(db_read_write_mutex_);
```

Reply:
I see, thanks for the explanation.  I've now changed it to `RCPPUTILS_TSA_REQUIRED`; please take another look.